python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -936,9 +936,9 @@ def __init__(
         state_dir: Optional[str] = None,
     ):
         if isinstance(project_dir, DbtProject):
-            populated_state_dir = project_dir.get_state_dir_if_populated()
-            if state_dir is None and populated_state_dir:
-                state_dir = os.fspath(populated_state_dir)
```

Comment:
Felt weird that the state directory had to be populated at definition. I'm thinking about the case where a manifest could be retrieved at run time.

Because of that, I'm removing this constraint. We should probably make `get_defer_args` throw an exception instead of returning an empty list of args.